### ISSUE TITLE
False-Positives: How do i limit? Zones? 

### ISSUE THREAD

**Issue by ozett** [jakowenko/double-take#91]
Is there a way to map out certain areas only for face-detetion?
or define zones? Like in frigate, but only in double-take for faces?
i have a lot of false positives of the cat feeder as a face ... 😹 

![image](https://user-images.githubusercontent.com/5937811/132309440-62d1ce08-ff66-4e92-8ee7-351fc8f88a5f.png)

**Comment by ozett** [comment]
maybe with the zones update 8 days ago ?
https://github.com/jakowenko/double-take/issues/79#issuecomment-907991482

**Comment by ozett** [comment]
i can try, but the image.size option dont seem clear to me.
in my double-take the mqtt-image.size is not defined. so it defaults to 500?
https://github.com/jakowenko/double-take#configuration

what size should i put in for mask? 
and i would define the mask in the frigate ui with the original cam-image dimension. 
does this work that way?

**Comment by jakowenko** [comment]
The size is for the stream that's coming into Frigate. In my case all my cameras are 1080p, so I used `1920x1080`.

This is needed to dynamically adjust the mask based on the size of the image Double Take is processing. You can define the mask in Frigate's UI, but you shouldn't need to save it there. But you can copy the values in Double Take.

If you want the mask applied to MQTT images, then you'll need to not crop the MQTT images in the Frigate `config.yml`. When they are cropped there is no way to know where the mask should be applied.

```yaml
mqtt:
  timestamp: False
  bounding_box: False
  crop: False
```

**Comment by ozett** [comment]
ok. i see.
i defined the image-dimension explicitly in frigate-config.yml to get it working without ffmpeg error.
so i will transfer that image-dimensions to the double-take config in try it ou. i come back with results...
thx

**Comment by ozett** [comment]
zones seem to work. 
glad that they show up !  👍 

![image](https://user-images.githubusercontent.com/5937811/132640592-71b8ec2c-e77b-4fbc-9495-b758c9285ac2.png)

**Comment by ozett** [comment]
filtering out face-detection with a mask seems not to work. what am i missing?
How could a face be detected, if i put a mask for non-detection over that region?

![image](https://user-images.githubusercontent.com/5937811/132739591-9187cb33-37ed-4be8-a986-dc02c70d4c64.png)

**Comment by jakowenko** [comment]
What was the "type" of that image on the left without the mask. If you have the mask visible and it's not shown in the image it means it wasn't applied to that one.

Is that the mqtt image from Frigate?

**Comment by ozett** [comment]
hmmm i can check it there is a difference, 
but i think all images for doubl-take come the same way from frigate. 
thats why i was wondering. i can check on detail tomorrow...

**Comment by jakowenko** [comment]
Just a heads up, I already told you about this in another issue haha, but I changed how masks are defined. I have a [sample](https://github.com/jakowenko/double-take/tree/beta#cameras) in the docs.

**Comment by ozett** [comment]
latest beta, new config-style.,     v1.0.0-482b7c9:beta

not shure if the mask still work... or it the new config-options broke it:
how to make the mask visible in the gui?

![image](https://user-images.githubusercontent.com/5937811/132949986-366521e9-caa4-4d22-ac47-9bf5a961285f.png)

**Comment by jakowenko** [comment]
I think I have the indentation wrong on the docs. I'll fix that.

The masks shouldn't be nested under `snapshot`.

```yaml
cameras:
  manual:
    masks:
      coordinates:
        - 1920,0,1920,328,1638,305,1646,0
        # - 0,1080,296,1080,322,827,0,835
      visible: true
      size: 1920x1080
```

**Comment by ozett** [comment]
> I think I have the indentation wrong on the docs. I'll fix that.

fix the docs? or the code?

should i change it like i thing it should be? 
or wait for the fixed docs?

i put me on hold and wait for your go 🚀

**Comment by jakowenko** [comment]
> > I think I have the indentation wrong on the docs. I'll fix that.
> 
> fix the docs? or the code?
> 
> should i change it like i thing it should be?
> or wait for the fixed docs?
> 
> i put me on hold and wait for your go 🚀

The code is working. Here's one I tested just now with this exact config. Notice the top right part of the image is masked out.

![Screen Shot 2021-09-11 at 10 48 10 AM](https://user-images.githubusercontent.com/1081811/132951755-aebd782d-e613-4257-b721-a97ab33f66c4.png)


```yaml
cameras:
  back-door:
    masks:
      coordinates:
        - 1920,0,1920,328,1638,305,1646,0
        # - 0,1080,296,1080,322,827,0,835
      visible: true
      size: 1920x1080
```

**Comment by ozett** [comment]
commented the snapshot-line out. now checking if that works..
```python
 cat-kueche-130:
    #snapshot:
      # apply masks before processing image
      masks:
        # list of x,y coordinates to define the polygon of the zone
        #- 3599,1896,1229,207,44,201,30,2118
        coordinates:
          #- 1920,0,1920,328,1638,305,1646,0
          - 1403,608,2485,1520,0,1520,664,592
        # show the mask on the final saved image (helpful for debugging)
        visible: true
        # size of camera stream used in resizing masks
        #size: 1920x1080
        #size: 3840x2160
        size: 2688x1520
```

**Comment by ozett** [comment]
no mask-info in the logs. i will check if its in the image.. (less identation maybe? did not throw an error..)
![image](https://user-images.githubusercontent.com/5937811/132952252-d575f95e-2faa-4b1c-b9d3-3f5c5c834ea5.png)

**Comment by jakowenko** [comment]
They won't show up in the response, but when you start Double Take you should see them in the logged out config.

![Screen Shot 2021-09-11 at 11 15 29 AM](https://user-images.githubusercontent.com/1081811/132952638-cbe7cf6e-6bef-4545-9537-2e098486e0da.png)

**Comment by jakowenko** [comment]
> commented the snapshot-line out. now checking if that works.. (can i use the http-api to upload testimages for mask-test?)
> 
> ```python
>  cat-kueche-130:
>     #snapshot:
>       # apply masks before processing image
>       masks:
>         # list of x,y coordinates to define the polygon of the zone
>         #- 3599,1896,1229,207,44,201,30,2118
>         coordinates:
>           #- 1920,0,1920,328,1638,305,1646,0
>           - 1403,608,2485,1520,0,1520,664,592
>         # show the mask on the final saved image (helpful for debugging)
>         visible: true
>         # size of camera stream used in resizing masks
>         #size: 1920x1080
>         #size: 3840x2160
>         size: 2688x1520
> ```

Make sure `masks:` is just indented once too. I think that will matter.

**Comment by ozett** [comment]
![image](https://user-images.githubusercontent.com/5937811/132953001-975db7a7-8e01-4d9f-a78a-5581cf8c18bd.png)
> Make sure masks: is just indented once too. I think that will matter.

Shift-Tab was my friend in the config-editor 😄

**Comment by jakowenko** [comment]
I'm going to close this since everything has been released in `1.0.0`. Feel fee to open a new issue if you run into problems.
